proposed/activity-improvements.md

```diff
@@ -0,0 +1,465 @@
+We are assuming that most Activity information is desirable by default, but the monitoring agent should be able to filter out noisy or problematic instrumentation. The agent can efficiently decide if it wants to listen to Activities which created from specific source/Component (ActivitySource).
+
+#### 3. Activity needs API representations of OpenTelemetry Span concepts
+
+What .Net calls Activity, OpenTelemetry calls Span. The OT span can express a broader set of information than Activity currently can. We want to create parity so that developers who want to emit certain OpenTelemetry data points can still use Activity APIs to accomplish that.
+
+#### 4. API usability
+
+Some of the existing patterns of API usage are awkward and don't appeal to many developers. We'd like to simplify usage wherever possible so that the APIs feel more approachable to a broader set of .Net developers.
+
+#### 5. Back compat
+
+Any scenarios that currently work using Activity and/or DiagnosticListener must continue to work with no code changes. Also the performance for these scenarios can't meaningfully regress. Our new Activity listening mechanism should be faster than conventional listening.
+
+### Non-goals
+
+1. Creating interfaces so that custom implementations of Activity can be created (OpenTelemetry does this with Span)
+2. Supporting isolated monitoring domains that allows one set of instrumentation producers and consumers to be segregated from others within the same process (OpenTelemetry can do this via multiple instances of the top-level Tracer object)
+3. Matching OpenTelemetry specification naming or behaviors that would directly conflict with .Net's existing precedent for Activity. We'll do our best to enable the use-cases OpenTelemetry is trying to achieve via some other means in these cases.
+
+
+Design
+=================
+
+[This is copied from Tarek's writeup in [the existing issue](https://github.com/dotnet/runtime/issues/31373) ]
+
+## Rationale and Use Cases
+
+The proposal here is mainly targeting closing the gap between the .NET Libraries and OpenTelemetry (OT) tracing APIs (mainly the OT Span class). We also enhance and simplify the usage of the Activity for easier code patterns. Here are the details of this proposal:
+
+- .NET Libraries have the `Activity` class which the object allows libraries and apps to publish tracing information. OT currently under development and not released yet. OT currently proposing the `Span` class which mostly used for the same purpose of Activity. OT Span supports more properties than Activity which allows more tracing scenarios. This proposal is offering the additions needed to be added to the Activity class which helps to address the same scenarios that Span does and possibly allow OT to fully get rid of Span to reduce the confusion which developers can run into to decide if need to use Activity or use Span. We are introducing the following types and properties which will be used inside the Activity class:
```

Comment:
Since OT was used for OpenTracing, OTel stuck more for OpenTelemetry:

```suggestion
- .NET Libraries have the `Activity` class which the object allows libraries and apps to publish tracing information. OTel currently under development and not released yet. OTel currently proposing the `Span` class which mostly used for the same purpose of Activity. OTel Span supports more properties than Activity which allows more tracing scenarios. This proposal is offering the additions needed to be added to the Activity class which helps to address the same scenarios that Span does and possibly allow OTel to fully get rid of Span to reduce the confusion which developers can run into to decide if need to use Activity or use Span. We are introducing the following types and properties which will be used inside the Activity class:
```